Struct with no fields is incorrectly treated as void

The following code fails to compile:
```zig
 const std = @import("std");
 const assert = std.debug.assert;
 const Thread = std.Thread;

 const Context = struct { };

 fn runWorker(ctx: Context) void {
     std.debug.print("in thread\n", .{});
 }

pub fn main() !void {
    var t = try Thread.spawn(runWorker, Context{});
    t.wait();
}
```
   
It gives the error:
```
./zig-linux-x86_64-0.8.0-dev.1825+d2d4df407/lib/std/Thread.zig:270:29: error: expected type 'Context', found 'void'
                    startFn(arg);
                            ^
./pool.zig:8:17: note: Context declared here
const Context = struct {};
```

[Link to godbolt](https://godbolt.org/z/b4xx3najb)

If I add a field to the struct, it [compiles](https://godbolt.org/z/6v6jcec81):

```zig
 const std = @import("std");
 const assert = std.debug.assert;
 const Thread = std.Thread;

 const Context = struct { foo: bool = false };

 fn runWorker(ctx: Context) void {
     std.debug.print("in thread\n", .{});
 }

pub fn main() !void {
    var t = try Thread.spawn(runWorker, Context{});
    t.wait();
}
```

This seems wrong and I asked in the #zig-help channel on discord and some people said it was likely a bug.

--- a/lib/std/Thread.zig
+++ b/lib/std/Thread.zig
@@ -199,7 +199,8 @@ pub fn spawn(comptime startFn: anytype, context: SpawnContextType(@TypeOf(startF
                 inner: Context,
             };
             fn threadMain(raw_arg: windows.LPVOID) callconv(.C) windows.DWORD {
-                const arg = if (@sizeOf(Context) == 0) {} else @ptrCast(*Context, @alignCast(@alignOf(Context), raw_arg)).*;
+                const arg = if (@sizeOf(Context) == 0) undefined //
+                else @ptrCast(*Context, @alignCast(@alignOf(Context), raw_arg)).*;
 
                 switch (@typeInfo(@typeInfo(@TypeOf(startFn)).Fn.return_type.?)) {
                     .NoReturn => {
@@ -260,7 +261,8 @@ pub fn spawn(comptime startFn: anytype, context: SpawnContextType(@TypeOf(startF
 
     const MainFuncs = struct {
         fn linuxThreadMain(ctx_addr: usize) callconv(.C) u8 {
-            const arg = if (@sizeOf(Context) == 0) {} else @intToPtr(*const Context, ctx_addr).*;
+            const arg = if (@sizeOf(Context) == 0) undefined //
+            else @intToPtr(*Context, ctx_addr).*;
 
             switch (@typeInfo(@typeInfo(@TypeOf(startFn)).Fn.return_type.?)) {
                 .NoReturn => {
@@ -292,7 +294,8 @@ pub fn spawn(comptime startFn: anytype, context: SpawnContextType(@TypeOf(startF
             }
         }
         fn posixThreadMain(ctx: ?*c_void) callconv(.C) ?*c_void {
-            const arg = if (@sizeOf(Context) == 0) {} else @ptrCast(*Context, @alignCast(@alignOf(Context), ctx)).*;
+            const arg = if (@sizeOf(Context) == 0) undefined //
+            else @ptrCast(*Context, @alignCast(@alignOf(Context), ctx)).*;
 
             switch (@typeInfo(@typeInfo(@TypeOf(startFn)).Fn.return_type.?)) {
                 .NoReturn => {

